### ISSUE TITLE
ruff-mutable-default (RUF008) not not catching mutable default passed to attrs.field.

### ISSUE THREAD

**Issue by ashb** [astral-sh/ruff#16495]
Simple repro case
```
❯ cat tst.py
import attrs

@attrs.define
class B :
    mutable_default: list[int] = []
    mutable_default2: list[int] = attrs.field(default=[])

❯ uvx ruff@0.9.9 check --extend-select RUF008,B006 --preview -n --isolated  tst.py
All checks passed!
```

_Originally posted by @ashb in https://github.com/astral-sh/ruff/issues/14327#issuecomment-2696968093_

Output:

```
tst.py:5:34: RUF008 Do not use mutable default values for dataclass attributes
  |
3 | @attrs.define
4 | class B :
5 |     mutable_default: list[int] = []
  |                                  ^^ RUF008
6 |     mutable_default2: list[int] = attrs.field(default=[])
  |
```

I was hoping it would catch the same error on `mutable_default2`

**Comment by MichaReiser** [comment]
Can you tell me more about what diagnostics you'd expect?

E.g. this raises RUF012 for the `mutable_default` field https://play.ruff.rs/98850596-5ef1-4a1e-a3e5-fa07ed757a81

**Comment by AlexWaygood** [comment]
I think this might only check `attrs` classes if preview mode is enabled. @ashb do you have `preview` mode enabled in your ruff config?

**Comment by MichaReiser** [comment]
> I think this might only check `attrs` classes if preview mode is enabled. [@ashb](https://github.com/ashb?rgh-link-date=2025-03-04T10%3A35%3A03.000Z) do you have `preview` mode enabled in your ruff config?

Preview mode is enabled, see the CLI command

**Comment by ashb** [comment]
🤦🏻 Okay, so that fixes the first one, but not the second. I guess it's up for debate if the second should be caught or not.

**Comment by MichaReiser** [comment]
Thanks for updating the summary. Adding support for `attrs.field` does make sense to me, considering that https://github.com/astral-sh/ruff/pull/14327#issuecomment-2696968093 introduced attrs support.

**Comment by fury0928** [comment]
Hi @ntBre I'd like to work on this issue if someone doesn't take up yet
Would you assign me for contribution?
Thank you
